java/src/main/java/ai/rapids/cudf/ColumnVector.java

```diff
@@ -3483,9 +3483,8 @@ private static NestedColumnVector createNestedColumnVector(DType type, long rows
             // least one byte or it will not be interpreted correctly.
             dataLen = 1;
           }
-        } else if (type.isNestedType()) {
-          dataLen = dataBuffer.length;
         }
+        dataLen = Math.min(dataLen,dataBuffer.length);
```

Comment:
What is the situation that caused this change?  This indicates that either someone is not giving us as much data as we expected, or our calculation is off.  Nested data types are the ones not truly covered here, and in what situation do we have a list, or a struct with an actual data column in it?

Reply:
The issue is `long dataLen = rows * type.sizeInBytes;` does not consider nulls, so we expect to copy more than what is actually there. Agree this is not really nested type related, in fact when we make the child vectors of base type with nulls, this becomes a problem. As I understand, we want to write values except the nulls.

Reply:
So this is for a child column of a list?  If so we need to have the correct number of rows for that child column, not for any parent column.

Reply:
I thought I had that working for rows of nested child column , I will check and fix. Thanks for the help!